### PR TITLE
VCS link improvements

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: downloader/src/main/kotlin/VcsHost.kt [hunk 1]
```diff
@@ -237,13 +237,15 @@ private fun toGitPermalink(
     vcsUrl: URI, revision: String, path: String, startLine: Int, endLine: Int,
     startLineMarker: String, endLineMarker: String
 ): String {
+    fun String.isPathToMarkdownFile() = endsWith(".md") || endsWith(".markdown")
```
Comment (fviernau) [inline downloader/src/main/kotlin/VcsHost.kt]: How about setting `ignorecase = true` at least for the former?

Also, consider moving this function scope to file level as the scope still is reasonable small while the reader of the outer function is not forced to have a look at the implementation.

File: reporter/src/main/kotlin/reporters/ReportTableModelMapper.kt [hunk 2]
```diff
@@ -113,9 +113,11 @@ class ReportTableModelMapper(private val resolutionProvider: ResolutionProvider)
                     it.summary.errors
                 }?.distinct() ?: emptyList()
 
+                val packageForId = ortResult.getPackage(id)?.pkg ?: ortResult.getProject(id)?.toPackage()
```
Comment (fviernau) [inline reporter/src/main/kotlin/reporters/ReportTableModelMapper.kt]: commit-msg: When reading this commit message the first questions popping up were
- `what else` is affected besides perma links
- is the change ok for other affected things

My understanding is that `nothing else` (beside perma links) is affected by this change.
Would you mind changing the commit message to make this more clear?

File: downloader/src/main/kotlin/VcsHost.kt (hunk 1 above)
Comment (sschuberth) [reply downloader/src/main/kotlin/VcsHost.kt]: > How about setting `ignorecase = true` at least for the former?

Actually, I believe it makes more sense for the latter, as I could imagine people writing e.g. `.MarkDown`, but I have never seen `.MD`. Done for both, anyway.

> Also, consider moving this function scope to file level

Originally, I didn't want to do this until the function is actually becoming useful elsewhere, but again done anyway.

File: reporter/src/main/kotlin/reporters/ReportTableModelMapper.kt (hunk 2 above)
Comment (sschuberth) [reply reporter/src/main/kotlin/reporters/ReportTableModelMapper.kt]: I hope to have made this clearer now.

Comment (fviernau) [reply reporter/src/main/kotlin/reporters/ReportTableModelMapper.kt]: Thanks!

**Pull request merged**